### ISSUE TITLE
failOnError: true not working for gradle plugin

### ISSUE THREAD

**Issue by EugenMayer** [dependency-check/DependencyCheck#4918]
**Describe the bug**
even though `failOnError: true` is defined and vuln. are found, the gradle task does not exit with an exit code > 0 

gradle configuration
```
dependencyCheck {
    formats = ["HTML", "JSON"]
    failOnError = true

    analyzers {
        assemblyEnabled = false
        nodeEnabled = false
    }
}
```

```
./gradlew :core:dependencyCheckAnalyze
```

```
Found 3 vulnerabilities in project core

One or more dependencies were identified with known vulnerabilities in core:

federation-graphql-java-support-2.0.0.jar (pkg:maven/com.apollographql.federation/federation-graphql-java-support@2.0.0, cpe:2.3:a:graphql-java_project:graphql-java:2.0.0:*:*:*:*:*:*:*) : CVE-2022-37734
federation-graphql-java-support-api-2.0.3.jar (pkg:maven/com.apollographql.federation/federation-graphql-java-support-api@2.0.3, cpe:2.3:a:graphql-java_project:graphql-java:2.0.3:*:*:*:*:*:*:*) : CVE-2022-37734
graphql-java-extended-scalars-18.2.jar (pkg:maven/com.graphql-java/graphql-java-extended-scalars@18.2, cpe:2.3:a:graphql-java_project:graphql-java:18.2:*:*:*:*:*:*:*) : CVE-2022-37734


See the dependency-check report for more details.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2s
8 actionable tasks: 2 executed, 6 up-to-date
```

```
echo $?                                                                                                               
0
```

**Version of dependency-check used**
7.2.1

**Log file**
(see above)

**Expected behavior**
if `failOnError = true` is used, exit code should be > 0 if any vuln are found

**Comment by aikebah** [comment]
Finding a vulnerability is not an error. For breaking a build when finding vulnerabilities there is the failBuildOnCVSS configuration. 

failOnError is to override the standard behavior that errors during analysis break the build.

https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/configuration.html

**Comment by EugenMayer** [comment]
@aikebah thank you for clarifying. I might at least suggest this could be improved in the docs. I'am not sure if it is to obvious that a dependency scanner, finding a vulnerability in task, should be considered well run.

Running tests has the very same analogy. A broken tests leads to exit code > 0 even though running the tests actually did not fail at all. I would assume a 'test failing' when running tests is at least comparable to a 'finding a vulnerability' when scanning for those. Both variants tests for 'all tests are passing' or 'all dependencies do pass as non vulnerable'.

Nevertheless, thank you for helping out with failBuildOnCVSS, this helps a lot

**Comment by aikebah** [comment]
I don't see how 

"Fails the build if **an error occurs during the dependency-check analysis**" (emphasis added)

can be unclear about its purpose, especially as the failBuildOnCVSS is directly above it in the documentation.

If it was meant to make it fail if a vulnerability was found it would've been named as such (failOnVulnerability)

Test runner frameworks are created as a 'build' plugin to validate the specification-adherence of your coding and as such consider test-failure "always build breaking" and offer a flag to disable that behavior. 
Once passing (and before that code should typically not be published for inclusion in the codebase) tests are expected to always pass, so a failing test signals a clear error in the codebase that makes it not ready to be integrated.

This project is designed as a reporting plugin (similar to automated code review plugins) and therefor does not break by default. This is explicitly documented in the failBuildOnCVSS option, and there is the parameter offered (similar to quality gates of code review plugins) to define from which CVSS level you consider a vulnerability bad enough to break your build rather than merely report on the vulnerability risks identified in your dependencies.

Vulnerabilities appear constantly, so the 'build that was working 3 seconds ago' could suddenly break (without any changes to the codebase) in the middle of the release process for everyone when we opted for your approach of break-by-default.

The report will still tell post release that you successfuly released with known vulnerabilities, and, depending on vulnerability mitigation policies, the severity of the vulnerability and the assessment of the exploitability of it in your codebase will define whether you would stop right there and fix it or continue and fix it in a later release. (Just like your code review tool will still report on issues in the codebase as long as quality gate thresholds are not broken)

Typically in the industry you see different solution/mitigation timeframes depending on the severity of a problem, typically defined in cut-off CVSS scores (a made-up sample regime could be. 9.0-10.0: fix right now, 7.0-8.9: fix within a week, 3.0-6.9: fix within 30 days, <3.0 fix at the latest in the next feature-release or 90 days whichever comes first). 
Only a certain range is in the 'halt all your current activities and fix this' category that should be build-breaking, and where that cut-off lies varies from company to company (in the sample policy that would be the 9.0 threshold)

**Comment by EugenMayer** [comment]
I think we have a different understanding here, but as you mention, there are different policies in different environments and companies and i guess neither of them is right or wrong, it just depends on a lot of factors.

So i guess it is more a discussion about the default what a failed dep. scan means, and what it does not. IMHO when i tell the process to fail on any errors, i would have expected that finding vuln. is part of that - but i understood your idea is, this is more of an report/overview plugin unless you configure a threshold that is lower then the score given. 
It might be also a discussion about the docs, but i guess, those depend, what expectations the reader has - and here we differ.

There is no reason to argue for me, i think what you say makes perfectly sense, it just is not my primary POV, but that is just fine. Using failBuildOnCVSS can be used to get what i need, and if needed, even more fine grained. 

So for me, this issue can be closed anytime you see fit. Thank you for the time explaining the idea behind it!
